From 9e798c47eac3f01295016b6f27a48e787a072c25 Mon Sep 17 00:00:00 2001
From: Manuel Nickschas <sputnick@quassel-irc.org>
Date: Thu, 28 Nov 2013 01:45:43 +0100
Subject: [PATCH] Make sys includes POSIX-compliant

We're using <sys/resources.h> in quassel.cpp for the backtrace generation
stuff. Documentation for that header indicates that <sys/time.h> shall also
be included (even though it doesn't seem to be needed for our usage).
Additionally, compiling started to fail in recent FreeBSD versions, because
we lacked the <sys/types.h> include, which seems to be mandated by POSIX
(but not on Linux, although it won't hurt to have it).

tl;dr: This should make Quassel compile on recent versions of FreeBSD again.
---
 src/common/quassel.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/common/quassel.cpp b/src/common/quassel.cpp
index e081775..058b4b9 100644
--- a/src/common/quassel.cpp
+++ b/src/common/quassel.cpp
@@ -23,6 +23,8 @@
 #include <iostream>
 #include <signal.h>
 #if !defined Q_OS_WIN32 && !defined Q_OS_MAC
+#  include <sys/types.h>
+#  include <sys/time.h>
 #  include <sys/resource.h>
 #endif
 
-- 
1.8.4

